runtime/j9vm/jvm.c

```diff
@@ -3998,6 +3998,50 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 }
 
 
+/**
+ * Validates the JNI library for offload.
+ * If successful, returns the file handle, otherwise calls JVM_UnloadLibrary to close the handle and returns NULL.
```

Comment:
This isn't quite right; it doesn't return the original handle.

Closing the library here seems a little drastic. However, with that behavior, I don't see any need to introduce a new function: the behavior of `JVM_LoadLibrary()` can be updated to achieve the same goal.

Reply:
> However, with that behavior, I don't see any need to introduce a new function: the behavior of JVM_LoadLibrary() can be updated to achieve the same goal.

Not sure what you are saying. The behaviour of JVM_ValidateJNILibrary() can't be done from JVM_LoadLibrary() because we don't know if it's a JNI library or not. Unless you are suggesting I modify the JVM_LoadLibrary parameters to pass in that info.